libs/langchain/langchain/utils/utils.py

```diff
@@ -133,3 +134,30 @@ def get_pydantic_field_names(pydantic_cls: Any) -> Set:
         if field.has_alias:
             all_required_field_names.add(field.alias)
     return all_required_field_names
+
+
+def build_extra_kwargs(
+    extra_kwargs: Dict[str, Any],
+    values: Dict[str, Any],
+    all_required_field_names: Set[str],
+) -> Dict[str, Any]:
+    """"""
+    for field_name in list(values):
+        if field_name in extra_kwargs:
+            raise ValueError(f"Found {field_name} supplied twice.")
+        if field_name not in all_required_field_names:
+            warnings.warn(
+                f"""WARNING! {field_name} is not default parameter.
+                {field_name} was transferred to model_kwargs.
+                Please confirm that {field_name} is what you intended."""
+            )
```

Comment:
I would be tempted to just raise in this case.

Reply:
i don't disagree but would be breaking change for OpenAI models, so maybe something to be done in a bigger release